### PR TITLE
Added BetterDiscord support with a plugin

### PR DESCRIPTION
**Pull request by diogotr7**
Adds support for discord status information. Uses only overlays, the regular layers are disabled by default.

Thanks to @Popat0 for writing most of the plugin.

### CODE REVIEW

File: Project-Aurora/Project-Aurora/Profiles/Discord/Control_Discord.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        private void InstallPlugin()
+        {
+            string appdata = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+            string bd = Path.Combine(appdata, "BetterDiscord", "plugins", "AuroraGSI.plugin.js");
+
+            if (!File.Exists(bd))            
+                Directory.CreateDirectory(Path.GetDirectoryName(bd));           
+
+            using (FileStream pluginStream = File.Create(Path.Combine(bd)))
```
Comment (Bluscream) [inline Project-Aurora/Project-Aurora/Profiles/Discord/Control_Discord.xaml.cs]: Why would you combine the already combined path again?

File: Project-Aurora/Project-Aurora/Profiles/Discord/DiscordApplication.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+namespace Aurora.Profiles.Discord
+{
+
+    public class Discord : Application {
+
+        public Discord() : base(new LightEventConfig
+        {
+            Name = "Discord",
+            ID = "discord",
+            ProcessNames = new[] { "Discord.exe", "DiscordPTB.exe" },
```
Comment (Bluscream) [inline Project-Aurora/Project-Aurora/Profiles/Discord/DiscordApplication.cs]: What about `DiscordCanary.exe` ?

File: Project-Aurora/Project-Aurora/Profiles/Discord/GSI/AuroraGSI.plugin.js [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    stop() {
+        clearInterval(this.updatetimer);
+		//this.unpatch();
+        this.ready = false;
+    }
+    
+	onLibLoaded() {
+        
+        this.settings = NeatoLib.Settings.load(this, this.defaultSettings);
+		NeatoLib.Updates.check(this, "https://raw.githubusercontent.com/Popat0/Discord-GSI/master/AuroraGSI.plugin.js");
```
Comment (Bluscream) [inline Project-Aurora/Project-Aurora/Profiles/Discord/GSI/AuroraGSI.plugin.js]: Why would you want another update check in the js code? Either use the pluginrepo from BD or do the update check here in C#

Comment (Popat0) [reply Project-Aurora/Project-Aurora/Profiles/Discord/GSI/AuroraGSI.plugin.js]: You do realize not everyone uses PluginRepo, right? As for Aurora it only is supposed to install the plugin for a first run, not check if it's up to date.

File: Project-Aurora/Project-Aurora/Profiles/Discord/DiscordApplication.cs (hunk 2 above)
Comment (diogotr7) [reply Project-Aurora/Project-Aurora/Profiles/Discord/DiscordApplication.cs]: Added that.

File: Project-Aurora/Project-Aurora/Profiles/Discord/Control_Discord.xaml.cs (hunk 1 above)
Comment (diogotr7) [reply Project-Aurora/Project-Aurora/Profiles/Discord/Control_Discord.xaml.cs]: This was probably a leftover from something i was doing earlier, just changed it.

**Comment by Bluscream** [comment]
OwO finally lul

**Comment by Bentechy66** [comment]
Plugin appears to be broken. Console output:
image

**Comment by Bentechy66** [comment]
> Plugin appears to be broken. Console output:
> image

Fixed

**Comment by Popat0** [comment]
> 
> Plugin appears to be broken. Console output:

Attempted a fix by updating the library a couple hours ago, unsure if it worked due to an acute case of lack of computer. Restarting Discord is enough to get the update.

**Pull request merged**